pkg/KubeArmorOperator/internal/controller/resources.go

```diff
@@ -0,0 +1,585 @@
+	enforcerVolumes, enforcerVolumeMounts := genEnforcerVolumes(enforcer)
+	runtimeVolumes, runtimeVolumeMounts := genRuntimeVolumes(runtime, socket, runtimeStorage)
+	vols := []corev1.Volume{}
+	volMnts := []corev1.VolumeMount{}
+	vols = append(vols, enforcerVolumes...)
+	vols = append(vols, runtimeVolumes...)
+	vols = append(vols, common.CommonVolumes...)
+	volMnts = append(volMnts, enforcerVolumeMounts...)
+	volMnts = append(volMnts, runtimeVolumeMounts...)
+	volMnts = append(volMnts, common.CommonVolumesMount...)
+
+	daemonset := deployments.GenerateDaemonSet("generic", common.Namespace)
+	daemonset.Name = name
+	labels := map[string]string{
+		common.EnforcerLabel:       enforcer,
+		common.RuntimeLabel:        runtime,
+		common.RuntimeStorageLabel: runtimeStorage,
+		common.SocketLabel:         socket,
+		common.OsLabel:             "linux",
+	}
+	daemonset.Spec.Template.Spec.NodeSelector = labels
+	daemonset.Spec.Template.Labels = labels
+	daemonset.Spec.Template.Spec.ServiceAccountName = "kubearmor"
+	daemonset.Spec.Selector = &metav1.LabelSelector{
+		MatchLabels: labels,
+	}
+	if deployment_uuid != "" {
+		daemonset.OwnerReferences = []metav1.OwnerReference{
+			{
+				APIVersion: "apps/v1",
+				Kind:       "Deployment",
+				Name:       deployment_name,
+				UID:        deployment_uuid,
+			},
+		}
+	}
+	daemonset.Spec.Template.Spec.Volumes = vols
+	daemonset.Spec.Template.Spec.InitContainers[0].VolumeMounts = common.CommonVolumesMount
+	daemonset.Spec.Template.Spec.Containers[0].VolumeMounts = volMnts
+	daemonset.Spec.Template.Spec.Containers[0].Args = append(daemonset.Spec.Template.Spec.Containers[0].Args, "-criSocket=unix:///"+strings.ReplaceAll(socket, "_", "/"))
+	daemonset = addOwnership(daemonset).(*appsv1.DaemonSet)
+	fmt.Printf("generated daemonset: %v", daemonset)
+	return daemonset
+}
+
+func genEnforcerVolumes(enforcer string) (vol []corev1.Volume, volMnt []corev1.VolumeMount) {
+	if enforcer == "none" {
+		return nil, nil
+	}
+	for _, e := range strings.Split(enforcer, ".") {
+		vol = append(vol, common.EnforcerVolumes[e]...)
+		volMnt = append(volMnt, common.EnforcerVolumesMounts[e]...)
+	}
+	return
+}
+
+func genRuntimeVolumes(runtime, runtimeSocket, runtimeStorage string) (vol []corev1.Volume, volMnt []corev1.VolumeMount) {
+	// lookup socket
+	for _, socket := range common.ContainerRuntimeSocketMap[runtime] {
+		if strings.ReplaceAll(socket[1:], "/", "_") == runtimeSocket {
+			vol = append(vol, corev1.Volume{
+				Name: runtime + "-socket",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: socket,
+						Type: &common.HostPathSocket,
+					},
+				},
+			})
+			volMnt = append(volMnt, corev1.VolumeMount{
+				Name:      runtime + "-socket",
+				MountPath: socket,
+				ReadOnly:  true,
+			})
+			break
+		}
+	}
+	// lookup runtime storage location
+
+	for _, storageLocation := range common.RuntimeStorageVolumes[runtime] {
```

Comment:
I'm using k3s with containerd (embedded) which has container storage at `/run/k3s/containerd` however it wasn't mounted on kubearmor. What might have happened? :thinking:

Reply:
it's handled